packages/composites/token-allocation/src/dataProvider.ts

```diff
@@ -27,31 +28,103 @@ import { Logger } from '@chainlink/ea-bootstrap'
 export const getPriceProvider =
   (source: string, jobRunID: string, apiConfig: RequestConfig) =>
   async (symbols: string[], quote: string, withMarketCap = false): Promise<ResponsePayload> => {
-    const results = await Promise.all(
-      symbols.map(async (base) => {
-        const data = {
-          id: jobRunID,
-          data: { base, quote, endpoint: withMarketCap ? 'marketcap' : 'crypto' },
-        }
-        try {
-          const response = await Requester.request({ ...apiConfig, data: data })
-          return response.data.result
-        } catch (error) {
-          Logger.error(`Request to ${source} adapter failed: ${error}`)
-          throw new Error(
-            `Failed to request the ${source} adapter. Ensure that the ${source.toUpperCase()}_ADAPTER_URL environment variable is correctly pointed to the adapter location.`,
-          )
-        }
-      }),
+    if (source === 'coinpaprika') {
+      return await sendBatchedRequests(source, jobRunID, apiConfig, symbols, quote, withMarketCap)
+    }
+    return await sendIndividualRequests(source, jobRunID, apiConfig, symbols, quote, withMarketCap)
+  }
+
+export interface BatchedAdapterResponse {
+  data: {
+    results: (AdapterResponse | number)[][]
```

Comment:
do we already have a type for batched responses?

Reply:
Yeah, it's in a weird place though. `packages/core/bootstrap/src/lib/modules/requester.ts`

Reply:
I can clean this up in my types PR.